### PR TITLE
🎨 Home Page Overview Section #9

### LINKED ISSUE #9: 🎨 Create Home Page Overview Section
#  Create the home Page "Overview" section

#### Note: This issue is dependent on #19 and #27 , please complete that before you continue.

Quick links: [Summary](#summary) | [Description](#description) | [Technical-Analysis](#Technical-Analysis) | [Acceptance-Criteria](#Acceptance-Criteria)

---

### Summary

As a user, I want a section that shows an Overview section that describes what Elewa does as shown in the image below

### Description

Design:

Info-section

Link to full design:

```
https://xd.adobe.com/view/b98caf27-6549-49e5-8ed8-c5234788df61-ecdf/screen/cc35e19a-24f2-42ce-8803-ee2942839528
```
### Technical Analysis

This feature can be found in libs/features/pages/home (already created)

### Acceptance Criteria

- [ ] When rendered the component is pixel-perfect compared to the design above.
- [ ] The component is clean and does not have more than 50 lines
- [ ] The component SCSS uses CSS variables for colour, font, ... Connected to the main style.scss
- [ ] The code is isolated to the libs/page/home package, unless a justified reason is specified.
- [ ] The component is aligned using CSS flex
- [ ] The design is responsive for screen, tablet and phone

### PR DESCRIPTION
**Pull request by preston-56**
# Description

- Created an `HomeOverviewSectionComponent` that describes what Elewa does.
- Reused the `image container` and `text content item` components from issues #19 and #27 respectively.
- The created overview section is responsive on different devices.

Fixes #9 

## Screenshot

# screen

Screenshot from 2023-09-11 14-46-31

# mobile

Screenshot from 2023-09-11 14-46-12

# Demo
mobile

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/data/sections/src/lib/home-overview-data.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// Define a separate function for the getLayoutClass method
+export function getLayoutClass(item: ImageAndText): string {
+  // Implement your logic to determine the layout class based on the item
+  // For example, you can access item.imagePosition and return a class name accordingly.
+  if (item && item.imagePosition) {
+    return item.imagePosition === 'left' ? 'left-image' : 'right-image';
+  }
+  // Return a default layout class if needed
+  return 'default-layout';
+}
```
Comment (LemmyMwaura) [inline libs/data/sections/src/lib/home-overview-data.ts:35]: remove this from here.

File: libs/features/pages/home/src/lib/main/home/home-page.component.html [hunk 2]
```diff
@@ -1,6 +1,14 @@
...
+        [imageConfig]="mapImageAndTextToImageConfig(item)"
+      ></elewa-website-image-container>
+    </div>
+    <div class="text">
+      <elewa-website-elewa-text-content-item
+        [content]="item.content"
+      ></elewa-website-elewa-text-content-item>
+    </div>
   </div>
-</elewa-website-main-page>
```
Comment (LemmyMwaura) [inline libs/features/pages/home/src/lib/main/home/home-page.component.html:6]: Why is this logic added here? It should be in a component in the components folder

File: libs/features/pages/home/src/lib/main/home/home-page.component.ts [hunk 3]
```diff
@@ -1,8 +1,32 @@
...
+    this.imageData = imageData;
+    }
+
+  mapImageAndTextToImageConfig(item: ImageAndText): ImageConfig {
+    return {
+      title: item.content.title,
+      imageSrc: item.image.imageSrc,
+      visualisation: item.image.visualisation || ImageVisualisation.Window, // Set a default value
+    };
+  }
```
Comment (LemmyMwaura) [inline libs/features/pages/home/src/lib/main/home/home-page.component.ts:26]: This is an unnecessary method

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
Make updates

File: libs/data/sections/src/lib/home-overview-data.ts (hunk 1 above)
Comment (preston-56) [reply libs/data/sections/src/lib/home-overview-data.ts:35]: @LemmyMwaura Does this suggestion translate to having(create) a component to handle this? The instructions are a bit vague.

Comment (LemmyMwaura) [reply libs/data/sections/src/lib/home-overview-data.ts:35]: Just have it as a method in your component that needs this.

Comment (preston-56) [reply libs/data/sections/src/lib/home-overview-data.ts:35]: @LemmyMwaura Thanks. That was not clear from the start.

Comment (preston-56) [reply libs/data/sections/src/lib/home-overview-data.ts:35]: @LemmyMwaura I have made an update of the component [i.e., I have created `section overview` component] and refactored the code to the required standard. Let me know if this update works as per requested changes. Thanks

File: libs/features/pages/home/src/lib/main/home/home-page.component.ts (hunk 3 above)
Comment (preston-56) [reply libs/features/pages/home/src/lib/main/home/home-page.component.ts]: @LemmyMwaura 

* resolved

File: libs/features/pages/home/src/lib/home-overview-section/home-overview-section.component.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+  ngOnInit() {
+    // Assign the imported data to imageData directly
+    this.imageData = imageData;
+  }
+
+  // Create a method to return ImageVisualisation enum
+  getImageVisualisation(): typeof ImageVisualisation {
+    return ImageVisualisation;
+  }
```
Comment (LemmyMwaura) [inline libs/features/pages/home/src/lib/home-overview-section/home-overview-section.component.ts:22]: ??. This method is redundant

File: libs/features/pages/home/src/lib/home-overview-section/home-overview-section.component.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  // Define the getLayoutClass function
+  getLayoutClass(item: ImageAndText): string {
+    // Implement your logic to determine the layout class based on the item
+    // For example, you can access item.imagePosition and return a class name accordingly.
+    if (item && item.imagePosition) {
+      return item.imagePosition === 'left' ? 'left-image' : 'right-image';
+    }
+    // Return a default layout class if needed
+    return 'default-layout';
```
Comment (LemmyMwaura) [inline libs/features/pages/home/src/lib/home-overview-section/home-overview-section.component.ts:31]:
```typeScript
 getLayoutClass(item: ImageAndText): string {
    return item.imagePosition === 'left' ? 'left-image' : 'right-image';
  }
  ```

Your if statement is redundant as the property is not optional

File: libs/features/pages/home/src/lib/main/home/home-page.component.html [hunk 6]
```diff
@@ -1,6 +1 @@
-<elewa-website-main-page>
-  <div mainPage class="home-page">
-    <elewa-website-projects-section></elewa-website-projects-section>
-    <elewa-website-news-section></elewa-website-news-section>
-  </div>
-</elewa-website-main-page>
```
Comment (LemmyMwaura) [inline libs/features/pages/home/src/lib/main/home/home-page.component.html:6]: This is important, Your component should just be added in its correct placement

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]

File: libs/features/pages/home/src/lib/home-overview-section/home-overview-section.component.ts (hunk 4 above)
Comment (preston-56) [reply libs/features/pages/home/src/lib/home-overview-section/home-overview-section.component.ts]: @LemmyMwaura ##resolved

File: libs/features/pages/home/src/lib/home-overview-section/home-overview-section.component.ts (hunk 5 above)
Comment (preston-56) [reply libs/features/pages/home/src/lib/home-overview-section/home-overview-section.component.ts]: @LemmyMwaura ##resolved

File: libs/features/pages/home/src/lib/main/home/home-page.component.html (hunk 6 above)
Comment (preston-56) [reply libs/features/pages/home/src/lib/main/home/home-page.component.html]: @LemmyMwaura ##resolved

File: libs/features/pages/home/src/lib/home-overview-section/home-overview-section.component.scss [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    display: flex;
+    flex-direction: row; /* Display items horizontally */
+    margin-bottom: 20px; /* Add margin between items */
+    margin-top: 5rem;
+    margin-right: 180px;
+  }
+  
+  .image {
+    flex: 1; /* Make the image take up 50% of the container */
+    padding: 0 10px; /* Add spacing around the image */
```
[the remark this answers is not in the document]
Comment (preston-56) [reply libs/features/pages/home/src/lib/home-overview-section/home-overview-section.component.scss:19]: @LemmyMwaura I seem not to understand what you mean here

[the remark this answers is not in the document]
Comment (preston-56) [reply libs/features/pages/home/src/lib/home-overview-section/home-overview-section.component.scss:19]: @LemmyMwaura I think I have seen the issue. Fixing asap!

**Comment by LemmyMwaura** [comment]
Closing this as it's been resolved

**Pull request closed without merging**